framework/quota/src/org/apache/cloudstack/quota/constant/QuotaTypes.java

```diff
@@ -0,0 +1,99 @@
+        quotaTypeList.put(Integer.valueOf(NETWORK_BYTES_RECEIVED), new QuotaTypes(Integer.valueOf(NETWORK_BYTES_RECEIVED), "NETWORK_BYTES_RECEIVED", "GB", "Network Usage (Bytes Received)"));
+        quotaTypeList.put(Integer.valueOf(VOLUME), new QuotaTypes(Integer.valueOf(VOLUME), "VOLUME", "GB-Month", "Volume Usage"));
+        quotaTypeList.put(Integer.valueOf(TEMPLATE), new QuotaTypes(Integer.valueOf(TEMPLATE), "TEMPLATE", "GB-Month", "Template Usage"));
+        quotaTypeList.put(Integer.valueOf(ISO), new QuotaTypes(Integer.valueOf(ISO), "ISO", "GB-Month", "ISO Usage"));
+        quotaTypeList.put(Integer.valueOf(SNAPSHOT), new QuotaTypes(Integer.valueOf(SNAPSHOT), "SNAPSHOT", "GB-Month", "Snapshot Usage"));
+        quotaTypeList.put(Integer.valueOf(SECURITY_GROUP), new QuotaTypes(Integer.valueOf(SECURITY_GROUP), "SECURITY_GROUP", "Policy-Month", "Security Group Usage"));
+        quotaTypeList.put(Integer.valueOf(LOAD_BALANCER_POLICY), new QuotaTypes(Integer.valueOf(LOAD_BALANCER_POLICY), "LOAD_BALANCER_POLICY", "Policy-Month", "Load Balancer Usage"));
+        quotaTypeList.put(Integer.valueOf(PORT_FORWARDING_RULE), new QuotaTypes(Integer.valueOf(PORT_FORWARDING_RULE), "PORT_FORWARDING_RULE", "Policy-Month", "Port Forwarding Usage"));
+        quotaTypeList.put(Integer.valueOf(NETWORK_OFFERING), new QuotaTypes(Integer.valueOf(NETWORK_OFFERING), "NETWORK_OFFERING", "Policy-Month", "Network Offering Usage"));
+        quotaTypeList.put(Integer.valueOf(VPN_USERS), new QuotaTypes(Integer.valueOf(VPN_USERS), "VPN_USERS", "Policy-Month", "VPN users usage"));
+        quotaTypeList.put(Integer.valueOf(VM_DISK_IO_READ), new QuotaTypes(Integer.valueOf(VM_DISK_IO_READ), "VM_DISK_IO_READ", "GB", "VM Disk usage(I/O Read)"));
+        quotaTypeList.put(Integer.valueOf(VM_DISK_IO_WRITE), new QuotaTypes(Integer.valueOf(VM_DISK_IO_WRITE), "VM_DISK_IO_WRITE", "GB", "VM Disk usage(I/O Write)"));
+        quotaTypeList.put(Integer.valueOf(VM_DISK_BYTES_READ), new QuotaTypes(Integer.valueOf(VM_DISK_BYTES_READ), "VM_DISK_BYTES_READ", "GB", "VM Disk usage(Bytes Read)"));
+        quotaTypeList.put(Integer.valueOf(VM_DISK_BYTES_WRITE), new QuotaTypes(Integer.valueOf(VM_DISK_BYTES_WRITE), "VPN_USERS", "GB", "VM Disk usage(Bytes Write)"));
+        quotaTypeList.put(Integer.valueOf(VM_SNAPSHOT), new QuotaTypes(Integer.valueOf(VM_SNAPSHOT), "VM_SNAPSHOT", "GB-Month", "VM Snapshot storage usage"));
+        quotaTypeList.put(Integer.valueOf(CPU_CLOCK_RATE), new QuotaTypes(Integer.valueOf(CPU_CLOCK_RATE), "CPU_CLOCK_RATE", "Compute-Month", "Quota tariff for using 1 CPU of clock rate 100MHz"));
+        quotaTypeList.put(Integer.valueOf(CPU_NUMBER), new QuotaTypes(Integer.valueOf(CPU_NUMBER), "CPU_NUMBER", "Compute-Month", "Quota tariff for running VM that has 1vCPU"));
+        quotaTypeList.put(Integer.valueOf(MEMORY), new QuotaTypes(Integer.valueOf(MEMORY), "MEMORY", "Compute-Month", "Quota tariff for using 1MB or RAM for 1 hour"));
```

Comment:
Why not provide ``QuotaTypes(String, String, String, String)`` constructor to wrap the perform the ``Integer.valueOf`` operation and remove the repeating group.